test/extended/router/externalCertificate.go

```diff
@@ -0,0 +1,682 @@
+						err = oc.KubeClient().RbacV1().RoleBindings(oc.Namespace()).Delete(context.Background(), SECRET_READER_ROLE_BINDING, metav1.DeleteOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						g.By("Re-creating the deleted secret")
+						_, err = oc.KubeClient().CoreV1().Secrets(oc.Namespace()).Create(context.Background(), secret, metav1.CreateOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+					})
+
+					g.It("then routes are not reachable", func() {
+						g.By("Checking the route status")
+						for i := 0; i < len(routes); i++ {
+							checkRouteStatus(oc, routes[i].Name, corev1.ConditionFalse, "ExternalCertificateValidationFailed")
+						}
+					})
+				})
+			})
+			g.Context("and the secret is updated", func() {
+				g.BeforeEach(func() {
+					g.By("Updating the existing secret")
+					// build a new secret
+					secret, rootDerBytes, err = createSecret(oc.Namespace(), "my-tls-secret", corev1.SecretTypeTLS, hosts...)
+					o.Expect(err).NotTo(o.HaveOccurred())
+					// update the existing secret with the new secret
+					_, err = oc.KubeClient().CoreV1().Secrets(oc.Namespace()).Update(context.Background(), secret, metav1.UpdateOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+				})
+				g.It("then also routes are reachable", func() {
+					g.By("Sending https request")
+					for i := 0; i < len(routes); i++ {
+						hostName, err := getHostnameForRoute(oc, routes[i].Name)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						resp, err := httpsGetCall(hostName, rootDerBytes)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						o.Expect(resp).Should(o.ContainSubstring("Hello OpenShift"))
+					}
+				})
+			})
+			g.Context("and the secret is updated but RBAC permissions are dropped", func() {
+				g.BeforeEach(func() {
+					g.By("Deleting RBAC permissions")
+					err = oc.KubeClient().RbacV1().RoleBindings(oc.Namespace()).Delete(context.Background(), SECRET_READER_ROLE_BINDING, metav1.DeleteOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					g.By("Updating the existing secret")
+					// build a new secret
+					secret, rootDerBytes, err = createSecret(oc.Namespace(), "my-tls-secret", corev1.SecretTypeTLS, hosts...)
+					o.Expect(err).NotTo(o.HaveOccurred())
+					// update the existing secret with the new secret
+					_, err = oc.KubeClient().CoreV1().Secrets(oc.Namespace()).Update(context.Background(), secret, metav1.UpdateOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+				})
+				g.It("then routes are not reachable", func() {
+					g.By("Checking the route status")
+					for i := 0; i < len(routes); i++ {
+						checkRouteStatus(oc, routes[i].Name, corev1.ConditionFalse, "ExternalCertificateValidationFailed")
+					}
+				})
+			})
+
+			g.Context("and the route is updated", func() {
+				var (
+					routeToTest   *routev1.Route
+					newSecretName = "new-ext-crt"
+				)
+
+				g.BeforeEach(func() {
+					// These tests donot *explicity* need verification on multiple routes.
```

Comment:
Okay, why the above ones need to be tested on multiple routes,
is only because we wanted to re-use the same secret being used by multiple Routes, right?

Reply:
I also have the same question. If we want to test on more than one route makes sense to use 2. If we are using 5, then it would make sense to use different configurations for them. I don't think we are expecting different behaviours for the different routes.